internal/storage/v1/elasticsearch/options.go

```diff
@@ -317,10 +338,27 @@ func (opt *Options) InitFromViper(v *viper.Viper) {
 }
 
 func initFromViper(cfg *namespaceConfig, v *viper.Viper) {
-	cfg.Authentication.BasicAuthentication.Username = v.GetString(cfg.namespace + suffixUsername)
-	cfg.Authentication.BasicAuthentication.Password = v.GetString(cfg.namespace + suffixPassword)
-	cfg.Authentication.BearerTokenAuthentication.FilePath = v.GetString(cfg.namespace + suffixTokenPath)
-	cfg.Authentication.BasicAuthentication.PasswordFilePath = v.GetString(cfg.namespace + suffixPasswordPath)
+	// BasicAuthentication if atleast one of username, password or passwordPath is set
+	username := v.GetString(cfg.namespace + suffixUsername)
+	password := v.GetString(cfg.namespace + suffixPassword)
+	passwordPath := v.GetString(cfg.namespace + suffixPasswordPath)
+
+	if username != "" || password != "" || passwordPath != "" {
+		cfg.Authentication.BasicAuthentication = configoptional.Some(config.BasicAuthentication{
+			Username:         username,
+			Password:         password,
+			PasswordFilePath: passwordPath,
+		})
+	}
+
+	// BearerTokenAuthentication if tokenPath is set
```

Comment:
Is this correct?
```suggestion
	// BearerTokenAuthentication if tokenPath is set.
	// The other fields of BearerTokenAuthentication are not supported via options.
```

Reply:
according to L414 it's not correct
```
bearerAuth.AllowFromContext = v.GetBool(bearertoken.StoragePropagationKey)
```
So why are you only checking for one field?